testsuite/forge/src/backend/local/swarm.rs

```diff
@@ -541,3 +546,41 @@ impl Swarm for LocalSwarm {
         todo!()
     }
 }
+
+#[derive(Debug)]
+pub struct ActiveNodesGuard {
+    counter: Arc<Mutex<usize>>,
+    slots: usize,
+}
+
+impl ActiveNodesGuard {
+    pub async fn grab(slots: usize, counter: Arc<Mutex<usize>>) -> Self {
+        let max = num_cpus::get();
```

Comment:
Its a bit confusing that this is not really the `max` number of nodes spawn as we will spawn more than max number if current guard is less than max and `slots + guard` > `max`.  May be let's find a better name instead of `max` or add a comment here to clarify that.

Reply:
cleaned up a bit